lindera/src/token_filter.rs

```diff
@@ -0,0 +1,428 @@
+use std::collections::HashSet;
+
+use lindera_core::token_filter::TokenFilter;
+use serde::{Deserialize, Serialize};
+
+use crate::{error::LinderaErrorKind, LinderaResult, Token};
+
+pub const STOP_WORDS_TOKEN_FILTER_NAME: &str = "stop_words";
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub struct StopWordsTokenFilterConfig {
+    stop_words: HashSet<String>,
+}
+
+impl StopWordsTokenFilterConfig {
+    pub fn new(stop_words: HashSet<String>) -> Self {
+        Self { stop_words }
+    }
+
+    pub fn from_slice(data: &[u8]) -> LinderaResult<Self> {
+        serde_json::from_slice(data).map_err(|err| LinderaErrorKind::Deserialize.with_error(err))
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct StopWordsTokenFilter {
+    config: StopWordsTokenFilterConfig,
+}
+
+impl StopWordsTokenFilter {
+    pub fn new(config: StopWordsTokenFilterConfig) -> Self {
+        Self { config }
+    }
+
+    pub fn from_slice(data: &[u8]) -> LinderaResult<Self> {
+        Ok(Self::new(StopWordsTokenFilterConfig::from_slice(data)?))
+    }
+}
+
+impl TokenFilter for StopWordsTokenFilter {
+    fn apply<'a>(&self, tokens: Vec<Token<'a>>) -> LinderaResult<Vec<Token<'a>>> {
+        let mut t = Vec::new();
+        for token in tokens.iter() {
+            if !self.config.stop_words.contains(token.text) {
+                t.push(token.clone());
+            }
+        }
+
+        Ok(t)
+    }
+}
+
+pub const LENGTH_TOKEN_FILTER_NAME: &str = "length";
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub struct LengthTokenFilterConfig {
+    min: Option<usize>,
+    max: Option<usize>,
+}
+
+impl LengthTokenFilterConfig {
+    pub fn new(min: Option<usize>, max: Option<usize>) -> Self {
+        Self { min, max }
+    }
+
+    pub fn from_slice(data: &[u8]) -> LinderaResult<Self> {
+        serde_json::from_slice(data).map_err(|err| LinderaErrorKind::Deserialize.with_error(err))
+    }
+}
+
+pub struct LengthTokenFilter {
+    config: LengthTokenFilterConfig,
+}
+
+impl LengthTokenFilter {
+    pub fn new(config: LengthTokenFilterConfig) -> Self {
+        Self { config }
+    }
+
+    pub fn from_slice(data: &[u8]) -> LinderaResult<Self> {
+        Ok(Self::new(LengthTokenFilterConfig::from_slice(data)?))
+    }
+}
+
+impl TokenFilter for LengthTokenFilter {
+    fn apply<'a>(&self, tokens: Vec<Token<'a>>) -> LinderaResult<Vec<Token<'a>>> {
+        let mut t = Vec::new();
+        for token in tokens.iter() {
+            let len = token.text.chars().count();
+            if let Some(min) = self.config.min {
+                if len < min {
+                    continue;
+                }
+            }
+
+            if let Some(max) = self.config.max {
+                if len > max {
+                    continue;
+                }
+            }
+
+            t.push(token.clone());
```

Comment:
We could take `Token`'s references to avoid `clone()`? Also, could we replace the `tokens` vector in place?

I've not tried but 
```
Vec<Token<'a>>
```

could be 

```
Vec<Rc<RefCell<<Token<'a>>>>
```

Reply:
I'll revise it to the no-copying method. Thanks.

Reply:
Fixed.
Please see the following commit:
https://github.com/lindera-morphology/lindera/pull/236/commits/388982ca25107c71aa0309d14384468dcbb503bd

Reply:
I have decided to use the retain function. What do you think?